src/base/index.vue

```diff
@@ -573,21 +690,41 @@ export default {
       // We check if this is running in the browser or not
       // because during SSR no cors preflight request is sent
       const inBrowser = typeof window !== 'undefined';
+
+      const queryStringObject = {}
+      if(Object.keys(this.$route.query).length !== 0) {
+        Object.keys(this.$route.query).map((queryKey) => {
+          queryStringObject[queryKey] = this.$route.query[queryKey]
+        })
+      }
+
       const config = {
         headers: sectionHeader(((inBrowser) ? {} : {origin: this.$sections.projectUrl})),
       };
       const URL =
         this.$sections.serverUrl +
         `/project/${this.$sections.projectId}/page/${this.pageName}`;
 
+      // TODO: Send Query string payload only when it is requested to be used (Use Sections ENV for it)
+      // {
+      //   "query_string": queryStringObject
+      // }
+
       axios
       .post(URL, {}, config)
       .then((res) => {
         const sections = res.data.sections;
+        this.allSections = res.data.sections;
         const views = {};
         sections.map((section) => {
           this.trackSectionComp(section.name, section.type);
-          if (section.settings) section.settings = JSON.parse(section.settings);
+          if (section.type === "configurable") {
+            section.settings = section.render_data[0].settings;
+            section.nameID = section.name;
+            section.name = section.name.split(":")[1];
```

Comment:
please add some comments in your code to explain more why are you doing what you are doing